synapse-modules/alkemio_room_control.py

```
@@ -0,0 +1,281 @@
+                logger.error(
+                    "Failed to notify adapter about DM request: %s -> %s, error: %s",
+                    initiator_user_id,
+                    target_user_id,
+                    str(e),
+                )
+                return False
+        return False
+
+    async def on_create_room(
+        self,
+        requester: "synapse.types.Requester",
```

Ruff (0.14.7)

Undefined name `synapse`

(F821)